### PR TITLE
Create the added services component

### PR DESCRIPTION
**Pull request by hatched**
Created and assembled the components necessary for the added services list component.

### CODE REVIEW

**Comment by hatched** [comment]
#### To QA

visit `/:flags:/react/deploy-target=bundle:openstack-elementry-31`
- No list should be visible if there are no services on the canvas.
- As the services and units are added to the canvas new line items should show up in the added services list.

File: jujugui/static/gui/src/app/components/added-services-list-item/test-added-services-list-item.js [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    shallowRenderer.render(
+        <juju.components.AddedServicesListItem
+          service={service} />);
+
+    var output = shallowRenderer.getRenderOutput();
+    assert.deepEqual(output.props.children, [
+      <img src="icon.gif" className="inspector-view__item-icon" />,
+      <span className="inspector-view__item-count">5</span>,' ' ,'demo',
+      undefined
+    ]);
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/added-services-list-item/test-added-services-list-item.js]: Note that the JSX elements are passed in an array, this is the 'hack' to get around them needing to be nested under a parent element. This means that items such as ' ' are actually included in the array of elements.

File: jujugui/static/gui/src/app/components/panel/test-panel.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    var services = []; // no services.
+
+    var shallowRenderer = testUtils.createRenderer();
+    shallowRenderer.render(
+        <juju.components.Panel
+          instanceName={instanceName}
+          services={services}/>);
+
+    var output = shallowRenderer.getRenderOutput();
+    assert.equal(output.props.className, 'panel-component ' + instanceName + ' hidden');
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/panel/test-panel.js]: `output.props` is contains the props that are defined on the outer element of the component
`output.props.children` are all of the children elements of the component.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/2873/
Test PASSed.

**Comment by huwshimi** [comment]
I think the proper QA url is "/:flags:/react/?deploy-target=bundle/openstack-telemetry-31"

**Comment by huwshimi** [comment]
Oops, the `_added-services-list-item.scss` file can be removed.

File: jujugui/static/gui/src/app/components/added-services-list-item/added-services-list-item.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+      top.size = top.size + '';
+      return top;
+    },
+
+    /**
+      Renders and returns the status icon if necessary.
+
+      @method _renderStatusIndicator
+      @param {Object} statusData The status data that will be used to generate
+        the staus icon.
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/added-services-list-item/added-services-list-item.js]: "sta**t**us"

**Comment by huwshimi** [comment]
Looks great @hatched!

At the point that the first service loads I get the following warning in my browser:

`Warning: Each child in an array or iterator should have a unique "key" prop. Check the render method of AddedServicesList. See https://fb.me/react-warning-keys for more information.`

Other than that QA OK :+1:.

**Comment by hatched** [comment]
Thanks for the review @huwshimi I've removed the scss file and fixed that warning. Each list item now has the service name as a key.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/2874/
Test FAILed.

File: jujugui/static/gui/src/app/components/added-services-list-item/test-added-services-list-item.js (hunk 1 above)
Comment (makyo) [inline jujugui/static/gui/src/app/components/added-services-list-item/test-added-services-list-item.js]: Not for this branch, but I almost wonder if we might want to create a react-aware set of assertion helpers that clarify this a bit better.  Again, later.

File: jujugui/static/gui/src/app/components/added-services-list/test-added-services-list.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
+'use strict';
```
Comment (makyo) [inline jujugui/static/gui/src/app/components/added-services-list/test-added-services-list.js]: Are we not putting license headers on tests?

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/2875/
Test PASSed.

**Comment by makyo** [comment]
Sweet, :+1: on code.  A few minor things that may fall out into slack cards if needed

File: jujugui/static/gui/src/app/components/added-services-list/test-added-services-list.js (hunk 4 above)
Comment (hatched) [inline jujugui/static/gui/src/app/components/added-services-list/test-added-services-list.js]: Oops, that was just an oversight, I'll add them.

**Comment by hatched** [comment]
Thanks for the reviews! :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/2876/
Test PASSed.

**Pull request merged**